adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1209,24 +1214,34 @@ protected final void onCommandReceived(
         // example: control/DEFAULT_TENANT/4711/req/xyz/light
         final String topic = String.format("%s/%s/%s/%s/%s/%s", subscription.getEndpoint(), tenantId, deviceId,
                 subscription.getRequestPart(), command.getRequestId(), command.getName());
+        LOG.debug("Publishing command to device [tenant-id: {}, device-id: {}, MQTT client-id: {}, QoS: {}]",
+                subscription.getTenant(), subscription.getDeviceId(), endpoint.clientIdentifier(), qos);
         endpoint.publish(topic, command.getPayload(), qos, false, false);
+        endpoint.publishAcknowledgeHandler(
```

Comment:
I think this should go into an `else if` block of the following `if` statement ...

Reply:
From the functionality perspective, it behaves the same if it is here or inside `else if` block. But I agree that we need not register the handler always, I will change it.